### PR TITLE
Cherry pick be data apps minus apps and pages

### PR DESCRIPTION
**Pull request by metamben**
This PR builds on #27341 and removes backend parts that are not planned for 0.46.

Anton's summary of what is needed and what is not needed:

- **Need** — Actions model (explicit and implicit), action-model association, CRUD and execution endpoints, etc.
- **Need** — Dashboard cards with actions attached
- **Don’t need** — App containers (aka app table), CRUD endpoints, changes to search/activity/recents to handle apps and pages, etc
- **Don’t need** — Dashboard’s is_app_page flag and related logic
- **Don’t need** — Scaffolding endpoints

New permissions related to action execution are kept, app permissions are of course gone.

Migrations are added to remove the `app` table, the `app_permission_graph_revision` table and the `report_dashboard.is_app_page` column.

### CODE REVIEW

File: src/metabase/api/action.clj [hunk 1]
```diff
@@ -36,63 +38,89 @@
...
-
 (api/defendpoint GET "/"
   "Returns cards that can be used for QueryActions"
   [model-id]
   {model-id su/IntGreaterThanZero}
-  (action/merged-model-action nil :card_id model-id))
+  (let [model (api/read-check Card model-id)]
+    ;; We don't check the permissions on the actions, we assume they are
+    ;; readable if the model is readable.
+    (action/actions-with-implicit-params [model] :model_id model-id)))
```
Comment (metamben) [inline src/metabase/api/action.clj:48]: This is an optimization. We don't check each action, we know that the rules is that they are readable if their model is readable.

File: src/metabase/api/activity.clj [hunk 2]
```diff
@@ -55,18 +54,10 @@
...
-   (when-let [dashboard-ids (get referenced-objects "dashboard")]
-     (let [id->page?                            (db/select-id->field :is_app_page Dashboard
-                                                                     :id [:in dashboard-ids])
-           {page-ids true dashboard-ids' false} (group-by (comp boolean id->page?)
-                                                          ;; only existing ids go back
-                                                          (keys id->page?))]
-       (cond-> {}
-         (seq page-ids)       (assoc "page" (set page-ids))
-         (seq dashboard-ids') (assoc "dashboard" (set dashboard-ids')))))
-   (into {} (for [[model ids] (dissoc referenced-objects "card" "dashboard")
```
Comment (metamben) [inline src/metabase/api/activity.clj:67]: There are no pages any more, so dashboards are handled like the other simple types at line 60.

File: src/metabase/api/activity.clj [hunk 3]
```diff
@@ -78,15 +69,12 @@
...
         existing-dataset? (partial model-exists? "dataset")
-        existing-page? (partial model-exists? "page")
         existing-card? (partial model-exists? "card")]
     (for [{:keys [model_id], :as activity} activities]
-      (let [model (cond
-                    (and (= (:model activity) "card")
-                         (existing-dataset? (:model_id activity))) "dataset"
-                    (and (= (:model activity) "dashboard")
-                         (existing-page? (:model_id activity)))    "page"
-                    :else                                          (:model activity))]
```
Comment (metamben) [inline src/metabase/api/activity.clj:89]: Ditto.

File: src/metabase/api/collection.clj [hunk 4]
```diff
@@ -754,16 +724,17 @@
...
-                     collection/root-collection)))
+                     (cond-> collection/root-collection
+                       collection-namespace (assoc :namespace collection-namespace)))))
 
 (defn create-collection!
   "Create a new collection."
   [{:keys [name color description parent_id namespace authority_level]}]
   ;; To create a new collection, you need write perms for the location you are going to be putting it in...
-  (write-check-collection-or-root-collection parent_id)
+  (write-check-collection-or-root-collection parent_id namespace)
```
Comment (metamben) [inline src/metabase/api/collection.clj:737]: This is a bugfix.

File: src/metabase/api/search.clj [hunk 5]
```diff
@@ -286,52 +281,28 @@
...
   (-> (base-query-for-model "collection" search-ctx)
-      (update :where (fn [where] [:and [(if (= model "app") :<> :=) :app.id nil] where]))
       (hh/left-join [CollectionBookmark :bookmark]
                     [:and
                      [:= :bookmark.collection_id :collection.id]
-                     [:= :bookmark.user_id api/*current-user-id*]]
-                    [App :app]
-                    [:= :app.collection_id :collection.id])
+                     [:= :bookmark.user_id api/*current-user-id*]])
       (add-collection-join-and-where-clauses :collection.id search-ctx)))
```
Comment (metamben) [inline src/metabase/api/search.clj:291]: This was shared between apps and collections (see the `defmethod`s below), now it's collection only.

File: src/metabase/api/search.clj [hunk 6]
```diff
@@ -286,52 +281,28 @@
...
-  (-> (base-query-for-model "dashboard" search-ctx)
-      (update :where (fn [where] [:and [:= :dashboard.is_app_page (= model "page")] where]))
+(s/defmethod search-query-for-model "dashboard"
+  [model search-ctx :- SearchContext]
+  (-> (base-query-for-model model search-ctx)
       (hh/left-join [DashboardBookmark :bookmark]
                     [:and
                      [:= :bookmark.dashboard_id :dashboard.id]
                      [:= :bookmark.user_id api/*current-user-id*]])
       (add-collection-join-and-where-clauses :dashboard.collection_id search-ctx)))
```
Comment (metamben) [inline src/metabase/api/search.clj:304]: Same thing, this was shared before.

File: src/metabase/models/collection/graph.clj [hunk 7]
```diff
@@ -69,6 +69,17 @@
...
+  "Return the permission graph for the collections with id in `collection-ids` and the root collection."
+  ([collection-ids] (collection-permission-graph collection-ids nil))
+  ([collection-ids collection-namespace]
+   (let [group-id->perms (group-id->permissions-set)]
+     {:revision (c-perm-revision/latest-id)
+      :groups   (into {} (for [group-id (db/select-ids PermissionsGroup)]
+                           {group-id (group-permissions-graph collection-namespace
+                                                              (group-id->perms group-id)
+                                                              collection-ids)}))})))
+
```
Comment (metamben) [inline src/metabase/models/collection/graph.clj:82]: Extracted from the function below.

File: src/metabase/query_processor/writeback.clj [hunk 8]
```diff
@@ -6,44 +6,15 @@
...
-                                           {:status-code 400
-                                            :type        qp.error-type/invalid-parameter
-                                            :parameters  parameters
-                                            :mappings    parameter-mappings})))]
-            (merge {:id     param-id
-                    :target target}
-                   param-value-info)))
-        parameters))
+    (f (f query rff context)) -> (f query rff context)"
+  [#'qp.perms/check-query-action-permissions])
```
Comment (metamben) [inline src/metabase/query_processor/writeback.clj:17]: Bad diff. `map-parameters` is gone, `execution-middleware` has been added.

File: test/metabase/test/util.clj [hunk 9]
```diff
@@ -822,6 +822,25 @@
...
+
+(defmacro with-all-users-permission
+  "Run `body` with the ''All Users'' group being granted the permission
+  specified by `permission-path`.
+
+  (mt/with-all-users-permission (perms/app-root-collection-permission :read)
+    ...)"
+  [permission-path & body]
+  `(do-with-all-users-permission ~permission-path (fn [] ~@body)))
+
```
Comment (metamben) [inline test/metabase/test/util.clj:843]: I think these are not in use, but Cam suggested to put them into util, so I think we can leave them.

**Review by snoe — APPROVED** [review]

**Pull request merged**